pkg/worker/workflow.go

```diff
@@ -234,28 +243,50 @@ func (w *worker) TriggerPipelineWorkflow(ctx workflow.Context, param *TriggerPip
 	}
 
 	if param.TriggerFromAPI {
-		if err := workflow.ExecuteActivity(ctx, w.PreTriggerActivity, &PreTriggerActivityParam{
-			WorkflowID:      workflowID,
-			SystemVariables: param.SystemVariables,
+		if err := workflow.ExecuteActivity(ctx, w.LoadRecipeActivity, &LoadRecipeActivityParam{
+			WorkflowID:         workflowID,
+			PipelineUID:        param.SystemVariables.PipelineUID,
+			PipelineReleaseUID: param.SystemVariables.PipelineReleaseUID,
 		}).Get(ctx, nil); err != nil {
 			return err
 		}
 	}
 
-	_ = workflow.ExecuteActivity(minioCtx, w.UploadRecipeToMinioActivity, &UploadRecipeToMinioActivityParam{
+	err := workflow.ExecuteActivity(minioCtx, w.UploadInputsToMinioActivity, &UploadInputsToMinioActivityParam{
 		PipelineTriggerID: param.SystemVariables.PipelineTriggerID,
 		ExpiryRuleTag:     param.SystemVariables.ExpiryRuleTag,
 	}).Get(ctx, nil)
+	if err != nil {
+		logger.Error("Failed to upload pipeline run input", zap.Error(err))
+	}
 
-	_ = workflow.ExecuteActivity(minioCtx, w.UploadInputsToMinioActivity, &UploadInputsToMinioActivityParam{
+	err = workflow.ExecuteActivity(minioCtx, w.UploadRecipeToMinioActivity, &UploadRecipeToMinioActivityParam{
 		PipelineTriggerID: param.SystemVariables.PipelineTriggerID,
 		ExpiryRuleTag:     param.SystemVariables.ExpiryRuleTag,
 	}).Get(ctx, nil)
+	if err != nil {
+		logger.Error("Failed to upload pipeline run recipe", zap.Error(err))
+	}
+
+	// TODO group everything under condition
```

Comment:
Sorry, this wasn't meant to be the final state of the changes. I had (staged but) uncommitted changes to upload the recipe and input only under the `TriggerFromAPI` condition. They were meant to be squashed into this commit but I added them separately in the next commit to ease the review.